libs/primitives/src/conversion.rs

```diff
@@ -67,3 +68,52 @@ pub fn fixed_point_to_balance<
 
 	new_integer_part.ensure_add(new_frac_part)
 }
+
+/// Converts a `uint` `Balance` of one precision into a `Balance` of another
+/// precision i.e:
+/// ```rust
+/// use cfg_primitives::conversion::convert_balance_decimals;
+/// # use frame_support::{assert_err, assert_ok};
+/// # use sp_runtime::DispatchError;
+/// # use sp_arithmetic::ArithmeticError;
+///
+/// assert_ok!(convert_balance_decimals(3u32, 6u32, 1_234_567u64), 1_234_567_000u64);
+/// assert_ok!(convert_balance_decimals(6u32, 3u32, 1_234_567u64), 1_234u64);
+/// assert_ok!(convert_balance_decimals(6u32, 6u32, 1_234_567u64), 1_234_567u64);
+///
+/// assert_err!(convert_balance_decimals(6u32, 42u32, 1_000_000u64), DispatchError::Arithmetic(ArithmeticError::Overflow));
+/// ```
+pub fn convert_balance_decimals<
+	Precision: AtLeast32BitUnsigned + TryInto<usize>,
+	Balance: BaseArithmetic + Copy,
+>(
+	from: Precision,
+	to: Precision,
+	balance: Balance,
+) -> Result<Balance, DispatchError> {
+	match from {
+		from if from == to => Ok(balance),
+		from if to > from => precision_diff::<Precision, Balance>(to, from)?
+			.ensure_mul(balance)
+			.map_err(DispatchError::from),
+		from => balance
+			.ensure_div(precision_diff::<Precision, Balance>(from, to)?)
+			.map_err(DispatchError::from),
+	}
+}
+
+fn precision_diff<
+	Precision: AtLeast32BitUnsigned + TryInto<usize>,
+	Balance: BaseArithmetic + Copy,
+>(
+	gt: Precision,
+	lt: Precision,
+) -> Result<Balance, DispatchError> {
+	ensure_pow(
+		Balance::from(10),
+		gt.ensure_sub(lt)?
+			.try_into()
+			.map_err(|_| DispatchError::Other("Unable to Convert decimal precision to u32"))?,
```

Comment:
FYI, there is an `ensure_into()` for this purpose

Reply:
Thanks, updated to use `ensure_into` :)